### PR TITLE
[benchmark] Add roman numeral conversion benchmark

### PR DESCRIPTION
**Pull request by lorentey**
This adds a minibenchmark for roman number conversion. Measures Substring slicing, Substring.starts(with:) and String.append(_:) with tiny String arguments.

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci please benchmark

**Review by milseman — APPROVED** [review]
LGTM

File: benchmark/single-source/RomanNumbers.swift [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+]
+
+extension BinaryInteger {
+  var romanNumeral: String {
+    var result = ""
+    var value = self
+  outer:
+    while value > 0 {
+      var position = 0
+      for i in position ..< romanTable.count {
```
Comment (airspeedswift) [inline benchmark/single-source/RomanNumbers.swift]: I'd prefer something like `for (i, (v, s) in romanTable.enumerated() {`

(or even `in zip(romanTable.indices, romanTable)`)

Whether that makes for a better benchmark is another matter :)

File: benchmark/single-source/RomanNumbers.swift [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+//
+
+public let RomanNumbers = [
+  BenchmarkInfo(
+    name: "RomanNumbers",
+    runFunction: run_RomanNumbers,
+    tags: [.api, .String, .algorithm])
+]
+
+let romanTable: [(Int, String)] = [
```
Comment (airspeedswift) [inline benchmark/single-source/RomanNumbers.swift]: You could also use DictionaryLiteral, which may be unbenchmarked currently (and has an ABI-stability bug to give a better name cos it's neither a dictionary nor a literal)

Comment (lorentey) [reply benchmark/single-source/RomanNumbers.swift]: Good idea; done! 

Lisp calls something close to this an "association list", which isn't a bad name. `AssociationList`? `MapList`/`MappingList`? `AssociativeArray`?

**Comment by lorentey** [comment]
@swift-ci please benchmark

**Comment by milseman** [comment]
@swift-ci please smoke test

**Comment by lorentey** [comment]
@swift-ci please smoke test

**Comment by lorentey** [comment]
@swift-ci please benchmark

**Pull request merged**